thumbor/detectors/animeface_detector/__init__.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# thumbor imaging service
+# https://github.com/thumbor/thumbor/wiki
+
+# Licensed under the MIT license:
+# http://www.opensource.org/licenses/mit-license
+# Copyright (c) 2011 globo.com thumbor@googlegroups.com
+
+# Fandom-change-start: add custom animeface_detector
+from typing import Dict
+
+from thumbor.detectors.local_detector import CascadeLoaderDetector
+
+HAIR_OFFSET = 0.12
+
+
+class Detector(CascadeLoaderDetector):
```

Comment:
I didn't find any logs logged out from our custom change in last 7 days. Let's simplify this filter. See  https://console.cloud.google.com/logs/query;query=%22Error%20during%20face%20detection%22;duration=P7D?referrer=search&project=fandom-prod
```
class Detector(CascadeLoaderDetector):

    def __init__(self, context, index, detectors):
        super(Detector, self).__init__(context, index, detectors)
        self.load_cascade_file(__file__, self.context.config.ANIME_FACE_DETECTOR_CASCADE_FILE)

    def __add_hair_offset(self, top, height):
        top = max(0, top - height * HAIR_OFFSET)
        return top

    def detect(self, callback):
        try:
            features = self.get_features()
        except Exception as e:
            logger.exception(e)
            logger.warn('Error during face detection; skipping to next detector')
            self.next(callback)
            return

        if features:
            for (left, top, width, height), neighbors in features:
                top = self.__add_hair_offset(top, height)
                self.context.request.focal_points.append(
                    FocalPoint.from_square(left, top, width, height, origin="Anime Face Detection")
                )
            callback()
        else:
            self.next(callback)
```